### PR TITLE
Link check on README.chplenv

### PR DESCRIPTION
**Pull request by lydia-duncan**
Fix a broken module link, replace some code segments on environment variables
with links to those sections, added a link to README.module_search in the
CHPL_MODULE_PATH description, and made a more specific link to README.building
in the CHPL_HOST_PLATFORM section for clarity (it was just the file, now it
goes to a section in the file).

### CODE REVIEW

**Comment by lydia-duncan** [comment]
@benharsh, would you mind reviewing please?

File: doc/release/README.chplenv.rst [hunk 1]
```diff
@@ -111,6 +111,8 @@ MANPATH
 Optional Settings
 -----------------
 
+.. _chpl_target_platform:
```
Comment (benharsh) [inline doc/release/README.chplenv.rst]: If all of the links to this section are within this file, I think you can just write "`CHPL_TARGET_PLATFORM`_" instead.

File: doc/release/README.chplenv.rst [hunk 2]
```diff
@@ -269,6 +273,8 @@ CHPL_MODULE_PATH
    directories containing the .chpl files named explicitly on the compiler
    command line.
 
+.. _chpl_locale_model:
```
Comment (benharsh) [inline doc/release/README.chplenv.rst]: See comment concerning CHPL_TARGET_PLATFORM's link.

**Comment by benharsh** [comment]
Other than those comments, looks good!

File: doc/release/README.chplenv.rst (hunk 1 above)
Comment (lydia-duncan) [inline doc/release/README.chplenv.rst]: Eh, I think it might be reasonable for other files to link to them, so I think I'll leave them as is

**Pull request merged**